pkg/cmd/admin/router/router.go

```diff
@@ -536,8 +548,13 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 		return fmt.Errorf("you must specify a service account for the router with --service-account")
 	}
 
-	if err := validateServiceAccount(kClient, namespace, cfg.ServiceAccount, cfg.HostNetwork); err != nil {
-		return fmt.Errorf("router could not be created; %v", err)
+	if err := validateServiceAccount(kClient, namespace, cfg.ServiceAccount, cfg.HostNetwork, cfg.HostPorts); err != nil {
+		err = fmt.Errorf("router could not be created; %v", err)
+		if !output {
+			return err
+		}
+		fmt.Fprintf(cmd.Out(), "error: %v\n", err)
+		defaultOutputErr = cmdutil.ErrExit
```

Comment:
this is so we still output all the objects in output mode, but also return an error code? won't writing the error to cmd.Out() mess up using the objects?